crates/protocol/src/block_info.rs

```diff
@@ -642,6 +754,56 @@ impl L1BlockInfoHolocene {
             eip_1559_elasticity,
         })
     }
+
+    /// Calculate the data gas for posting the transaction on L1. Calldata costs 16 gas per byte
+    /// after compression.
+    pub fn data_gas(&self, input: &[u8]) -> U256 {
```

Comment:
This also doesn't seem like it'll work - The `Ecotone` variant needs to support the fjord tx cost function, which is before this variant will be active.

Reply:
Yeah duplicated the `ecotone` method below because of this. Let me look into how to de-dup these methods.